### PR TITLE
HV: modularization to separate CR related code

### PR DESCRIPTION
**Pull request by mcao6**
move the CR related code from vmcs/vcpu to vCR source files.

Tracked-On: #1842
Signed-off-by: Minggui Cao <minggui.cao@intel.com>
Reviewed-by: Jason Chen CJ <jason.cj.chen@intel.com>

### CODE REVIEW

File: hypervisor/include/arch/x86/virtual_cr.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+/* CR4 bits hv want to trap to track status change */
+#define CR4_TRAP_MASK (CR4_PSE | CR4_PAE | CR4_VMXE | CR4_PCIDE)
+#define	CR4_RESERVED_MASK ~(CR4_VME | CR4_PVI | CR4_TSD | CR4_DE | CR4_PSE | \
+				CR4_PAE | CR4_MCE | CR4_PGE | CR4_PCE |     \
+				CR4_OSFXSR | CR4_PCIDE | CR4_OSXSAVE |       \
+				CR4_SMEP | CR4_FSGSBASE | CR4_VMXE |         \
+				CR4_OSXMMEXCPT | CR4_SMAP | CR4_PKE |        \
+				CR4_SMXE | CR4_UMIP)
+
```
Comment (dongyaozu) [inline hypervisor/include/arch/x86/virtual_cr.h]: This 2 MACROs are only used by virtul_cr.c, should we move it to .c file?

Comment (mcao6) [reply hypervisor/include/arch/x86/virtual_cr.h]: right, will check again. thanks!

**Comment by wenlingz** [comment]
@mcao6 please fix your code, thanks

**Comment by mcao6** [comment]
updated for "violations to the coding".

**Comment by HuiHuangShi** [comment]
2364<br>16 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:sgdt&nbsp;&nbsp;offset:280:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : sgdt<br><br>2.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:sidt&nbsp;&nbsp;offset:287:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : sidt<br><br>3.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:get_cpu_id&nbsp;&nbsp;offset:416:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : get_cpu_id<br><br>4.filename:/hypervisor/include/arch/x86/msr.h&nbsp;&nbsp;function:pat_mem_type_invalid&nbsp;&nbsp;offset:411:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : pat_mem_type_invalid<br><br>5.filename:/hypervisor/include/arch/x86/vmcs.h&nbsp;&nbsp;function:get_vcpu_mode&nbsp;&nbsp;offset:55:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : get_vcpu_mode<br><br>6.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vmx_write_cr0&nbsp;&nbsp;offset:109:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Function return value potentially unused. : cr0_changed_bits<br><br>7.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vmx_write_cr4&nbsp;&nbsp;offset:262:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Function return value potentially unused. : old_cr4<br><br>8.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr0&nbsp;&nbsp;offset:342:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>9.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr0&nbsp;&nbsp;offset:347:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Construct leads to infeasible code.<br><br>10.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr0&nbsp;&nbsp;offset:355:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>11.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr2&nbsp;&nbsp;offset:360:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : vcpu_get_cr2<br><br>12.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr2&nbsp;&nbsp;offset:365:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : vcpu_set_cr2<br><br>13.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr4&nbsp;&nbsp;offset:370:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>14.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr4&nbsp;&nbsp;offset:375:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Construct leads to infeasible code.<br><br>15.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr4&nbsp;&nbsp;offset:383:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>16.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:cr_access_vmexit_handler&nbsp;&nbsp;offset:388:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : cr_access_vmexit_handler<br><br>

File: hypervisor/arch/x86/virtual_cr.c [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+ *   - CD (30) Trapped to emulate cache disable situation
+ *   - PG (31) Trapped to track cpu/paging mode.
+ *             Set the value according to the value from guest.
+ */
+static void vmx_write_cr0(struct acrn_vcpu *vcpu, uint64_t cr0)
+{
+	uint64_t cr0_vmx;
+	uint32_t entry_ctrls;
+	bool old_paging_enabled = is_paging_enabled(vcpu);
+	uint64_t cr0_changed_bits = vcpu_get_cr0(vcpu) ^ cr0;
```
Comment (HuiHuangShi) [inline hypervisor/arch/x86/virtual_cr.c]: can this line put into else branch?

File: hypervisor/arch/x86/virtual_cr.c [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+ *         We always keep align with physical cpu. So it's flexible to
+ *         guest
+ *   - SMEP (20) Flexible to guest
+ *   - SMAP (21) Flexible to guest
+ *   - PKE (22) Flexible to guest
+ */
+static void vmx_write_cr4(struct acrn_vcpu *vcpu, uint64_t cr4)
+{
+	uint64_t cr4_vmx, cr4_shadow;
+	uint64_t old_cr4 = vcpu_get_cr4(vcpu);
```
Comment (HuiHuangShi) [inline hypervisor/arch/x86/virtual_cr.c]: can this line put into else branch?

File: hypervisor/arch/x86/virtual_cr.c [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+	}
+	return ctx->cr0;
+}
+
+void vcpu_set_cr0(struct acrn_vcpu *vcpu, uint64_t val)
+{
+	vmx_write_cr0(vcpu, val);
+}
+
+uint64_t vcpu_get_cr2(const struct acrn_vcpu *vcpu)
```
Comment (HuiHuangShi) [inline hypervisor/arch/x86/virtual_cr.c]: Does vcpu_get_cr2 be called or referenced in other places?

**Comment by mcao6** [comment]
update as Huihuang's comments

**Comment by HuiHuangShi** [comment]
2364<br>14 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:sgdt&nbsp;&nbsp;offset:280:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : sgdt<br><br>2.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:sidt&nbsp;&nbsp;offset:287:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : sidt<br><br>3.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:get_cpu_id&nbsp;&nbsp;offset:416:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : get_cpu_id<br><br>4.filename:/hypervisor/include/arch/x86/msr.h&nbsp;&nbsp;function:pat_mem_type_invalid&nbsp;&nbsp;offset:411:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : pat_mem_type_invalid<br><br>5.filename:/hypervisor/include/arch/x86/vmcs.h&nbsp;&nbsp;function:get_vcpu_mode&nbsp;&nbsp;offset:55:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : get_vcpu_mode<br><br>6.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr0&nbsp;&nbsp;offset:342:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>7.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr0&nbsp;&nbsp;offset:347:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Construct leads to infeasible code.<br><br>8.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr0&nbsp;&nbsp;offset:355:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>9.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr2&nbsp;&nbsp;offset:360:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : vcpu_get_cr2<br><br>10.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr2&nbsp;&nbsp;offset:365:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : vcpu_set_cr2<br><br>11.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr4&nbsp;&nbsp;offset:370:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>12.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr4&nbsp;&nbsp;offset:375:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Construct leads to infeasible code.<br><br>13.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr4&nbsp;&nbsp;offset:383:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>14.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:cr_access_vmexit_handler&nbsp;&nbsp;offset:388:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : cr_access_vmexit_handler<br><br>

**Comment by HuiHuangShi** [comment]
1-14 are false positive

**Comment by mcao6** [comment]
update to doc issue.

**Comment by HuiHuangShi** [comment]
2364<br>14 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:sgdt&nbsp;&nbsp;offset:280:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : sgdt<br><br>2.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:sidt&nbsp;&nbsp;offset:287:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : sidt<br><br>3.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:get_cpu_id&nbsp;&nbsp;offset:416:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : get_cpu_id<br><br>4.filename:/hypervisor/include/arch/x86/msr.h&nbsp;&nbsp;function:pat_mem_type_invalid&nbsp;&nbsp;offset:554:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : pat_mem_type_invalid<br><br>5.filename:/hypervisor/include/arch/x86/vmcs.h&nbsp;&nbsp;function:get_vcpu_mode&nbsp;&nbsp;offset:55:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : get_vcpu_mode<br><br>6.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr0&nbsp;&nbsp;offset:342:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>7.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr0&nbsp;&nbsp;offset:347:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Construct leads to infeasible code.<br><br>8.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr0&nbsp;&nbsp;offset:355:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>9.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr2&nbsp;&nbsp;offset:360:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : vcpu_get_cr2<br><br>10.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr2&nbsp;&nbsp;offset:365:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : vcpu_set_cr2<br><br>11.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr4&nbsp;&nbsp;offset:370:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>12.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr4&nbsp;&nbsp;offset:375:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Construct leads to infeasible code.<br><br>13.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr4&nbsp;&nbsp;offset:383:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>14.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:cr_access_vmexit_handler&nbsp;&nbsp;offset:388:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : cr_access_vmexit_handler<br><br>

**Comment by mcao6** [comment]
please hold on, will resolve the doc failure.

**Comment by HuiHuangShi** [comment]
2364<br>14 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:sgdt&nbsp;&nbsp;offset:280:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : sgdt<br><br>2.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:sidt&nbsp;&nbsp;offset:287:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : sidt<br><br>3.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:get_cpu_id&nbsp;&nbsp;offset:416:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : get_cpu_id<br><br>4.filename:/hypervisor/include/arch/x86/msr.h&nbsp;&nbsp;function:pat_mem_type_invalid&nbsp;&nbsp;offset:554:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : pat_mem_type_invalid<br><br>5.filename:/hypervisor/include/arch/x86/vmcs.h&nbsp;&nbsp;function:get_vcpu_mode&nbsp;&nbsp;offset:55:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : get_vcpu_mode<br><br>6.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr0&nbsp;&nbsp;offset:342:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>7.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr0&nbsp;&nbsp;offset:347:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Construct leads to infeasible code.<br><br>8.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr0&nbsp;&nbsp;offset:355:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>9.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr2&nbsp;&nbsp;offset:360:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : vcpu_get_cr2<br><br>10.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr2&nbsp;&nbsp;offset:365:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : vcpu_set_cr2<br><br>11.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr4&nbsp;&nbsp;offset:370:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>12.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr4&nbsp;&nbsp;offset:375:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Construct leads to infeasible code.<br><br>13.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr4&nbsp;&nbsp;offset:383:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>14.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:cr_access_vmexit_handler&nbsp;&nbsp;offset:388:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : cr_access_vmexit_handler<br><br>

**Comment by mcao6** [comment]
just checked, it should be OK now. the failure not related with this patch.

**Comment by HuiHuangShi** [comment]
2364<br>14 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:sgdt&nbsp;&nbsp;offset:280:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : sgdt<br><br>2.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:sidt&nbsp;&nbsp;offset:287:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : sidt<br><br>3.filename:/hypervisor/include/arch/x86/cpu.h&nbsp;&nbsp;function:get_cpu_id&nbsp;&nbsp;offset:416:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : get_cpu_id<br><br>4.filename:/hypervisor/include/arch/x86/msr.h&nbsp;&nbsp;function:pat_mem_type_invalid&nbsp;&nbsp;offset:554:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : pat_mem_type_invalid<br><br>5.filename:/hypervisor/include/arch/x86/vmcs.h&nbsp;&nbsp;function:get_vcpu_mode&nbsp;&nbsp;offset:55:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Static procedure is not explicitly called in code analysed. : get_vcpu_mode<br><br>6.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr0&nbsp;&nbsp;offset:342:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>7.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr0&nbsp;&nbsp;offset:347:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Construct leads to infeasible code.<br><br>8.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr0&nbsp;&nbsp;offset:355:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>9.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr2&nbsp;&nbsp;offset:360:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : vcpu_get_cr2<br><br>10.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr2&nbsp;&nbsp;offset:365:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : vcpu_set_cr2<br><br>11.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr4&nbsp;&nbsp;offset:370:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>12.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_get_cr4&nbsp;&nbsp;offset:375:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Construct leads to infeasible code.<br><br>13.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:vcpu_set_cr4&nbsp;&nbsp;offset:383:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>14.filename:/hypervisor/arch/x86/virtual_cr.c&nbsp;&nbsp;function:cr_access_vmexit_handler&nbsp;&nbsp;offset:388:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : cr_access_vmexit_handler<br><br>

**Pull request merged**